src/bundles/peer-locations.js

```diff
@@ -12,243 +13,76 @@ export default function (opts) {
-          // Re-queue the peer if it has another address to try
-          queuingPeers: hasAlternate
-            ? state.queuingPeers.concat(peerId)
-            : state.queuingPeers,
-          locations: {
-            ...state.locations,
-            [peerId]: {
-              ...state.locations[peerId],
-              [addr]: {
-                state: 'failed',
-                error: action.payload.error
-              }
-            }
-          }
-        }
-      }
-
-      return state
+    actionBaseType: 'PEER_LOCATIONS',
+    getPromise: async ({ store, getIpfs }) => {
+      const peers = store.selectPeers()
+      return peerLocResolver.findLocations(peers, getIpfs)
     },
-
-    // Returns an object of the form:
-    // { [peerId]: { [multiaddr]: { state, data?, error? } } }
-    selectPeerLocationsRaw: state => state.peerLocations.locations,
-
-    // Select just the data for the peer locations that have been resolved
-    // Returns an object of the form:
-    // { [peerId]: { /* location data */ } }
-    selectPeerLocations: createSelector(
-      'selectPeerLocationsRaw',
-      peerLocsRaw => Object.keys(peerLocsRaw).reduce((locs, peerId) => {
-        const locsByAddr = peerLocsRaw[peerId]
-        const addr = Object.keys(locsByAddr).find(a => locsByAddr[a].state === 'resolved')
-        if (addr) locs[peerId] = peerLocsRaw[peerId][addr].data
-        return locs
-      }, {})
-    ),
-
-    selectPeerLocationsQueuingPeers: state => state.peerLocations.queuingPeers,
-
-    selectPeerLocationsResolvingPeers: state => state.peerLocations.resolvingPeers,
-
-    selectPeerLocationsForSwarm: createSelector(
-      'selectPeers',
-      'selectPeerLocations',
-      'selectBootstrapPeers',
-      (peers, locations, bootstrapPeers) => peers && peers.map(peer => {
-        const peerId = peer.peer.toB58String()
-        const locationObj = locations[peerId]
-        const location = toLocationString(locationObj)
-        const flagCode = locationObj && locationObj.country_code
-        const coordinates = locationObj && [
-          locationObj.longitude,
-          locationObj.latitude
-        ]
-        const connection = parseConnection(peer.addr)
-        const address = peer.addr.toString()
-        const latency = parseLatency(peer.latency)
-        const notes = parseNotes(peer, bootstrapPeers)
-
-        return {
-          peerId,
-          location,
-          flagCode,
-          coordinates,
-          connection,
-          address,
-          latency,
-          notes
-        }
-      })
-    ),
-
-    selectPeerCoordinates: createSelector(
-      'selectPeerLocationsForSwarm',
-      peers => {
-        if (!peers) return []
-        const allCoord = peers
-          .map(p => p.coordinates)
-          .filter(arr => !!arr)
-
-        const unique = new Set(allCoord.map(JSON.stringify))
-        return Array.from(unique).map(JSON.parse)
+    staleAfter: ms.seconds(1),
+    retryAfter: ms.seconds(1),
```

Comment:
Would be good to have the interval as a const at top of file, and documented

Reply:
Just added!